Docs: Update debian.md

oddlittlebird opened the pull request:
**What this PR does / why we need it**:
Update Debian/Ubuntu installation instructions to not use add-apt-repository, since it doesn't work uniformly across all distributions (e.g. Ubuntu 16.04).

**Which issue(s) this PR fixes**:

Fixes #22527

**Special notes for your reviewer**:

oddlittlebird committed:
Update debian.md

aknuds1 commented on docs/sources/installation/debian.md:
```diff
@@ -39,13 +39,6 @@ If you install from the APT repository, then Grafana is automatically updated ev
 sudo apt-get install -y apt-transport-https
 sudo apt-get install -y software-properties-common wget
 wget -q -O - https://packages.grafana.com/gpg.key | sudo apt-key add -
-
-
-# Alternatively you can add the beta repository, see in the table above
-sudo add-apt-repository "deb https://packages.grafana.com/enterprise/deb stable main"
-
-sudo apt-get update
-sudo apt-get install grafana-enterprise
```
Why do you want to remove instructions for installing beta versions?

aknuds1 commented on docs/sources/installation/debian.md:
```diff
@@ -54,12 +47,6 @@ sudo apt-get install grafana-enterprise
 sudo apt-get install -y apt-transport-https
 sudo apt-get install -y software-properties-common wget
 wget -q -O - https://packages.grafana.com/gpg.key | sudo apt-key add -
-
-# Alternatively you can add the beta repository, see in the table above
-sudo add-apt-repository "deb https://packages.grafana.com/oss/deb stable main"
-
-sudo apt-get update
-sudo apt-get install grafana
```
Why do you want to remove instructions for installing beta versions?

sakjur reviewed (dismissed):
This PR removes the installation step entirely, in #22527 the proposed solution is to replace

```bash
sudo add-apt-repository "deb https://packages.grafana.com/enterprise/deb stable main"
```

with

```bash
echo "deb https://packages.grafana.com/oss/deb stable main" | sudo tee -a /etc/apt/sources.list.d/grafana.list
```

rather than removing any lines in these instructions.

Lines 65-79 can be removed after that is done, since they're alternative instructions for when `add-apt-repository` fails.

aknuds1 reviewed (changes requested):
Having read the linked issue, I think the changes in this PR are based on a misunderstanding. The current changes _remove_ instructions for installing Grafana on Debian/Ubuntu, while the linked issue says to change them slightly.

I.e., the wanted change is from `sudo add-apt-repository "deb https://packages.grafana.com/enterprise/deb stable main"` to `echo "deb https://packages.grafana.com/oss/deb stable main" | sudo tee -a /etc/apt/sources.list.d/grafana.list `.

aknuds1 commented:
@sakjur is right. I think along with the alternate section (which will now be the main solution), we need to remove the heading "Set up repository for ARM" also, since it's just out of place.

xlson commented:
What @sakjur and @aknuds1 said. We want to replace the add-apt-rep line with the echo ... line and then remove the arm section as that will be exactly the same instruction as for ubuntu/debian.

oddlittlebird committed:
Revert "Update debian.md"

This reverts commit 063c1f09d9e490bd514bbd17b84c8bec38379295.

oddlittlebird committed:
Update debian.md

oddlittlebird commented:
Okeydokey. I reverted the deletion and changed the one line, but I am confused about what to do with the Set up repository for ARM section. Does it stay? Does part of it get deleted? Moved?

sakjur commented:
@oddlittlebird You can remove the entire section, it seems to refer to the ARM version of the repo having a problem when you're using the `add-apt-repository` command (I'm guessing, the section title is a bit confusing as the instructions are not specific for ARM), so the instructions in that section are now the same as the general instructions above.

oddlittlebird committed:
Update debian.md

oddlittlebird commented:
If I remove the entire section, then we lose the echo command for the beta repository. How's this?

aknuds1 reviewed (changes requested):
Looks better, but the same changes should be made for the enterprise section also.

oddlittlebird committed:
Update debian.md

aknuds1 commented on docs/sources/installation/debian.md:
````diff
@@ -40,12 +40,22 @@ sudo apt-get install -y apt-transport-https
+```
 sudo apt-get update
-sudo apt-get install grafana-enterprise
+sudo apt-get install grafana
````
This change should be undone
```suggestion
sudo apt-get install grafana-enterprise
```

oddlittlebird committed:
Update docs/sources/installation/debian.md

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 reviewed (approved):
LGTM
